cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/validator/AwsIDBrokerMappedRolePermissionValidator.java

```diff
@@ -0,0 +1,151 @@
+
+    /**
+     * Checks if the location should be checked or not for role permissions
+     * @param location StorageLocationBase to evaluate
+     * @return true if location is valid to check false otherwise
+     */
+    abstract boolean checkLocation(StorageLocationBase location);
+
+    /**
+     * Validates the cloudFileSystem
+     * @param cloudFileSystem cloud file system to evaluate
+     * @param validationResultBuilder builder for any errors encountered
+     */
+    public void validate(AmazonIdentityManagement iam, CloudS3View cloudFileSystem,
+                            ValidationResultBuilder validationResultBuilder) {
+        AccountMappingBase accountMappings = cloudFileSystem.getAccountMapping();
+        if (accountMappings != null) {
+            SortedSet<String> roleArns = getRoleArnsForUsers(getUsers(), accountMappings.getUserMappings());
+            Set<Role> roles = AwsIamUtil.getValidRoles(iam, roleArns, validationResultBuilder);
+
+            boolean s3guardEnabled = cloudFileSystem.getS3GuardDynamoTableName() != null;
+            List<String> policyFileNames = getPolicyFileNames(s3guardEnabled);
+
+            SortedSet<String> failedActions = new TreeSet<>();
+            for (StorageLocationBase location : cloudFileSystem.getLocations()) {
+                if (checkLocation(location)) {
+                    Map<String, String> replacements = getPolicyJsonReplacements(location, cloudFileSystem);
+                    List<Policy> policies = getPolicies(policyFileNames, replacements);
+                    for (Role role : roles) {
+                        List<EvaluationResult> evaluationResults = AwsIamUtil.validateRolePolicies(iam,
+                            role, policies);
+                        failedActions.addAll(getFailedActions(role, evaluationResults));
+                    }
+                }
+            }
+            if (!failedActions.isEmpty()) {
+                validationResultBuilder.error(String.format("The role(s) (%s) don't have the required permissions:%n%s",
+                    String.join(", ", roles.stream().map(Role::getArn).collect(Collectors.toCollection(TreeSet::new))),
+                    String.join("\n", failedActions)));
+            }
+        }
+    }
+
+    /**
+     * Get the replacements necessary for the policy json
+     * @param location StorageLocationBase to get storage paths from
+     * @param cloudFileSystem CloudFileSystem to get dynamodb table name
+     * @return map of simple replacements for policy json
+     */
+    protected Map<String, String> getPolicyJsonReplacements(StorageLocationBase location,
+                                                            CloudS3View cloudFileSystem) {
+        String storageLocationBase = getStorageLocationBase(location);
+        String datalakeBucket = storageLocationBase.split("/", 2)[0];
+        Map<String, String> replacements = new HashMap<>();
```

Comment:
Minor: A builder construct like `Map.of(Map.entry(key, value), ...)` would be more elegant / modern, provided the return value of `Map.put()` is irrelevant and an immutable map is acceptable.